subsys/spm/spm.c

```diff
@@ -382,7 +384,7 @@ static void spm_configure_ns(const tz_nonsecure_setup_conf_t
 	/* Allow SPU to have precedence over (non-existing) ARMv8-M SAU. */
 	tz_sau_configure(0, 1);
 
-#if defined(CONFIG_ARMV7_M_ARMV8_M_FP)
+#if defined(CONFIG_ARMV7_M_ARMV8_M_FP) && CONFIG_SPM_NRF_FPU_NS
 	/* Allow Non-Secure firmware to use the FPU */
 	SCB->NSACR |=
```

Comment:
Please, see if you can use the upstream API function for that: void tz_nonsecure_fpu_access_enable(void);